### PR TITLE
Feature/riak kv pb index dialyzer

### PR DESCRIPTION
**Pull request by Vagabond**
2 dialyzer fixes. The change to riak_kv_index.hrl seems right to us (Reid and I) but the riak_kv_index code is so hard to follow we can't prove it.

cc @engelsanchez and @russelldb as people with relevant experience.

### CODE REVIEW

File: src/riak_kv_pb_index.erl [hunk 1]
```diff
@@ -107,8 +107,6 @@ process(#rpbindexreq{} = Req, State) ->
             maybe_perform_query(QueryVal, Req, State)
     end.
 
-maybe_perform_query({error, Reason}, _Req, State) ->
```
Comment (engelsanchez) [inline src/riak_kv_pb_index.erl:110]: Why is this being removed? It would be good to list the dialyzer warnings you are trying to fix in the PR to follow along.

**Comment by Vagabond** [comment]
Relevant dialyzer warning:

```
riak_kv_pb_index.erl:93: Clause guard cannot succeed. The pattern {'ok', {'riak_kv_index_v3', _, _, Start, _, _, _, _, _, Re, _}} was matched against the type {'error',{'field_parsing_failed',{_,_}} | {'unknown_field_t
ype',binary()} | {'downgrade_not_supported',_,{_,_,_} | {_,_,_,_} | {_,_,_,_,_,_,_,_,_} | {_,_,_,_,_,_,_,_,_,_,_}} | {'invalid_continuation',{_,_},{_,_,_,_,_,_,_,_,_,_,_}}} | {'ok',{'eq',binary(),'undefined' | binary()
} | {'range',binary(),'undefined' | binary(),'undefined' | binary()} | #riak_kv_index_v2{start_key::binary(),filter_field::binary(),start_term::'undefined' | binary(),end_term::'undefined' | binary(),return_terms::bool
ean(),start_inclusive::boolean(),end_inclusive::boolean(),return_body::boolean()} | #riak_kv_index_v3{start_key::binary(),filter_field::binary(),start_term::'undefined' | binary(),end_term::'undefined' | binary(),retur
n_terms::boolean(),start_inclusive::boolean(),end_inclusive::boolean(),return_body::boolean(),term_regex::'undefined' | binary(),max_results::'undefined' | integer()}}
riak_kv_pb_index.erl:110: The pattern <{'error', Reason}, _Req, State> can never match the type <{'ok',{'eq',binary(),'undefined' | binary()} | {'range',binary(),'undefined' | binary(),'undefined' | binary()} | #riak_k
v_index_v2{start_key::binary(),filter_field::binary(),start_term::'undefined' | binary(),end_term::'undefined' | binary(),return_terms::boolean(),start_inclusive::boolean(),end_inclusive::boolean(),return_body::boolean
()} | #riak_kv_index_v3{start_key::binary(),filter_field::binary(),start_term::'undefined' | binary(),end_term::'undefined' | binary(),return_terms::boolean(),start_inclusive::boolean(),end_inclusive::boolean(),return_
body::boolean(),term_regex::'undefined' | binary(),max_results::'undefined' | integer()}},#rpbindexreq{term_regex::'undefined' | binary() | maybe_improper_list(binary() | maybe_improper_list(any(),binary() | []) | byte(),binary() | [])},_>
```

**Comment by russelldb** [comment]
Such tact. Wow.

File: src/riak_kv_pb_index.erl (hunk 1 above)
Comment (Vagabond) [inline src/riak_kv_pb_index.erl:110]: Look at the commit message, it explains that the check done in process() makes this dead code.

**Comment by russelldb** [comment]
There are so many ways to talk to people at Basho, being publicly blunt and dismissive while doing general tidy up on existing code is possibly the least productive. I get that the mission you and reid is on is holy and great, but don't steam roller people in the process, please.

**Comment by Vagabond** [comment]
Sorry, I phrased that poorly. I meant that I was unable to follow the code to understand what the types are expected to be in this particular case and was looking for someone with expert knowlesge to confirm that 2i lookups for integers are indeed stored as integers in the record.

Again, apologies for words written in haste, regretted at leisure.

**Comment by engelsanchez** [comment]
Ok. This looks correct to me.  :+1:

**Comment by reiddraper** [comment]
@borshop merge please
